### PR TITLE
Fix XXPlusYYGate documentation

### PR DESCRIPTION
**Pull request by kevinsung**
### Summary
The sign of both the phases and the exponential were off. I've added a test that shows the correct formula.

### Details and comments
Apparently these have been wrong since XYGate was first introduced. Unfortunately, I used this incorrect formula as the basis for XXMinusYYGate, and the result is that the formulas for XXPlusYYGate and XXMinusYYGate differ in sign in more places than I originally expected.

### CODE REVIEW

**Comment by kevinsung** [comment]
Upon internal discussion it appears the docs are correct but the implementation is incorrect. Marking as draft to prevent merging until we decide what to do.

File: qiskit/circuit/library/standard_gates/xx_plus_yy.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class XXPlusYYGate(Gate):
         \newcommand{\th}{\frac{\theta}{2}}
 
         R_{XX+YY}(\theta, \beta)\ q_0, q_1 =
-          RZ_1(\beta) \cdot \exp\left(-i \frac{\theta}{2} \frac{XX+YY}{2}\right) \cdot RZ_1(-\beta) =
+          RZ_1(-\beta) \cdot \exp\left(i \frac{\theta}{2} \frac{XX+YY}{2}\right) \cdot RZ_1(\beta) =
             \begin{pmatrix}
                 1 & 0                     & 0                    & 0  \\
                 0 & \cos\left(\th\right)             & i\sin\left(\th\right)e^{i\beta} & 0  \\
```
Comment (jakelishman) [inline qiskit/circuit/library/standard_gates/xx_plus_yy.py:48]: Regardless of the result of the discussion about how the gate _should_ be defined, I did this maths myself and agree with the PR's version matrix expressions.

**Review by jakelishman — APPROVED** [review]
Looks good to me.  Thanks for pushing through the decision quagmire!

**Review by jakelishman — APPROVED** [review]

**Comment by jakelishman** [comment]
btw, to follow on from your question in the meeting around how the backport works from the repo perspective:
- the `main` branch is the development version of the next minor release (so currently it's 0.21)
- all contributor PRs are made against `main`, so by default the first time they'll be released is the next minor
- we have separate branches for each minor series, so the current stable branch is `stable/0.20`
- if a PR is tagged with the "stable backport potential" label, then when it merges to `main`, the mergify bot opens a new PR to `stable/0.20` that's a cherry-pick of the commit
- we approve and merge that, and then when it comes time to release 0.20.1, we tag a commit on the `stable/0.20` branch, _not_ `main`.

The most recent example of the mergify bot backporting is #7955 if you're interested.

**Pull request merged**